### PR TITLE
Add operator annotations

### PR DESCRIPTION
**Pull request by howardjohn**
for https://github.com/istio/istio/issues/18621

### CODE REVIEW

File: annotation/annotations.yaml [hunk 1]
```diff
@@ -312,3 +312,21 @@ annotations:
...
+    resources:
+      - Any
+  - name: install.operator.istio.io/owner-generation
+    description: Represents the generate to which the resource was last reconciled.
+    deprecated: false
+    hidden: false
+    resources:
+      - Any
+  - name: install.operator.istio.io/version
+    description: Represents the Istio version associate with the resource
```
Comment (geeknoid) [inline annotation/annotations.yaml]:
```suggestion
    description: Represents the Istio version associated with the resource
```

File: annotation/annotations.yaml [hunk 2]
```diff
@@ -312,3 +312,21 @@ annotations:
...
     resources:
       - Ingress
+  - name: install.operator.istio.io/chart-owner
+    description: Represents the name of the chart used to create this resource.
+    deprecated: false
+    hidden: false
+    resources:
+      - Any
+  - name: install.operator.istio.io/owner-generation
+    description: Represents the generate to which the resource was last reconciled.
```
Comment (geeknoid) [inline annotation/annotations.yaml]: The generate?

Comment (howardjohn) [reply annotation/annotations.yaml]: I can't type :slightly_smiling_face: fixed

**Comment by sushicw** [comment]
Would it be possible to add `func All() []*annotation.Instance` to `annotations.gen.go`? This would allow us to update the annotations analyzer to not use a hand-maintained list.

**Comment by howardjohn** [comment]
> Would it be possible to add func All() []*annotation.Instance to annotations.gen.go? This would allow us to update the annotations analyzer to not use a hand-maintained list.

 probably so, sound useful. I think the code is in the tools repo

File: annotation/annotations.yaml [hunk 3]
```diff
@@ -312,3 +312,21 @@ annotations:
...
+    resources:
+      - Any
+  - name: install.operator.istio.io/owner-generation
+    description: Represents the generation to which the resource was last reconciled.
+    deprecated: false
+    hidden: false
+    resources:
+      - Any
+  - name: install.operator.istio.io/version
+    description: Represents the Istio version associated with the resource
```
Comment (rshriram) [inline annotation/annotations.yaml]: nit: full stop missing.

**Pull request merged**